examples/vision/image_captioning.py

```diff
@@ -0,0 +1,510 @@
+                else:
+                    caption_mapping[img_name] = [caption]
+
+        return caption_mapping, text_data
+
+
+def train_val_split(caption_data, train_size=0.8, shuffle=True):
+    """Split the captioning dataset into train and validation sets.
+
+    Args:
+        caption_data (dict): Dictionary containing the mapped caption data
+        train_size (float): Fraction of all the full dataset to use as training data
+        shuffle (bool): Whether to shuffle the dataset before splitting
+
+    Returns:
+        Traning and validation datasets as two separated dicts
+    """
+
+    # 1. Get the list of all image names
+    all_images = list(caption_data.keys())
+
+    # 2. Shuffle if necessary
+    if shuffle:
+        np.random.shuffle(all_images)
+
+    # 3. Split into training and validation sets
+    train_size = int(len(caption_data) * train_size)
+
+    training_data = {
+        img_name: caption_data[img_name] for img_name in all_images[:train_size]
+    }
+    validation_data = {
+        img_name: caption_data[img_name] for img_name in all_images[train_size:]
+    }
+
+    # 4. Return the splits
+    return training_data, validation_data
+
+
+# Load the dataset
+captions_mapping, text_data = load_captions_data("Flickr8k.token.txt")
+
+# Split the dataset into training and validation sets
+train_data, valid_data = train_val_split(captions_mapping)
+print("Number of training samples: ", len(train_data))
+print("Number of validation samples: ", len(valid_data))
+
+"""
+## Vectorizing the text data
+
+We'll use the `TextVectorization` layer to vectorize the text data,
+that is to say, to turn the
+original strings into integer sequences where each integer represents the index of
+a word in a vocabulary. We will use a custom string standardization scheme
+(strip punctuation characters except `<` and `>`) and the default
+splitting scheme (split on whitespace).
+"""
+
+
+def custom_standardization(input_string):
+    lowercase = tf.strings.lower(input_string)
+    return tf.strings.regex_replace(lowercase, "[%s]" % re.escape(strip_chars), "")
+
+
+strip_chars = "!\"#$%&'()*+,-./:;<=>?@[\]^_`{|}~"
+strip_chars = strip_chars.replace("<", "")
+strip_chars = strip_chars.replace(">", "")
+
+vectorization = TextVectorization(
+    max_tokens=VOCAB_SIZE,
+    output_mode="int",
+    output_sequence_length=SEQ_LENGTH,
+    standardize=custom_standardization,
+)
+vectorization.adapt(text_data)
+
+"""
+## Building a tf dataset pipeline for training
+
+We will generate pairs of images and the corresponding captions using `tf.data.Dataset`
+The pipeline consists of two steps:
+
+1. Read the image from the disk
+2. Tokenize all the five captions corresponding to this image
+"""
+
+
+def read_image(img_path, size=IMAGE_SIZE):
+    img = tf.io.read_file(img_path)
+    img = tf.image.decode_jpeg(img, channels=3)
+    img = tf.image.resize(img, IMAGE_SIZE)
+    return img
+
+
+def text_to_seq(cap):
```

Comment:
Also you don't really need this utility function since you can just do `.map(vectorization)` (saves you 2 lines of code)

Reply:
Ah yes!